### DISCUSSION TITLE
snapshot：why create a temp image and then create snapshot on the temp image？

### DISCUSSION THREAD

**Discussion by maydaycc** [ceph/ceph-csi#2729] [Q&A]
Current design of snapshot：
parent image ->create temp snapshot on parent image -> clone a temp image from temp snapshot -> delete temp snapshot ->create snapshot on temp image

Since increasing the clone depth will reduce the read performance, why not create a snapshot directly on the parent image？

**Comment by nixpanic** [comment]
I would expect the reason for this procedure to be documented in https://github.com/ceph/ceph-csi/blob/devel/docs/design/proposals/rbd-snap-clone.md . I don't see it immediately though :disappointed:

**Comment by Madhu-1** [comment]
it was done to keep snapshots and clones independent of each other and also to flatten the temporary cloned image, not the actual image which will be mounted by application. flattening rbd images given to users will have an i/o impact. you can configure the depth of the images using https://github.com/ceph/ceph-csi/blob/cf6fdcb796471603cd87382842a652bcd81f98a9/cmd/cephcsi.go#L105-L114 if you want each image to be independent try configuring these values to lower one. This will also solve the max limit of 510 snapshots on the rbd image.
